pkg/logql/error.go

```diff
@@ -0,0 +1,70 @@
+package logql
+
+import (
+	"fmt"
+
+	"github.com/grafana/loki/pkg/logql/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+// ParseError is what is returned when we failed to parse.
+type ParseError struct {
+	msg       string
+	line, col int
+}
+
+func (p ParseError) Error() string {
+	if p.col == 0 && p.line == 0 {
+		return fmt.Sprintf("parse error : %s", p.msg)
+	}
+	return fmt.Sprintf("parse error at line %d, col %d: %s", p.line, p.col, p.msg)
+}
+
+func newParseError(msg string, line, col int) ParseError {
+	return ParseError{
+		msg:  msg,
+		line: line,
+		col:  col,
+	}
+}
+
+func newStageError(expr Expr, err error) ParseError {
+	return ParseError{
+		msg:  fmt.Sprintf(`stage '%s' : %s`, expr, err),
+		line: 0,
+		col:  0,
+	}
+}
+
+// IsParseError returns true if the err is a ast parsing error.
+func IsParseError(err error) bool {
+	_, ok := err.(ParseError)
```

Comment:
we could use new `errors.Is` api :)
https://golang.org/pkg/errors/#Is
Two advantages: 
1. avoid type assertion
2. can also be wrapped errors!

Reply:
Definitively a good idea. I would prefer to keep this for another PR I'm chasing after a bug for 2.0.

Reply:
of course :) np!